### ISSUE TITLE
Issue #26: Integrate TFLite for Document Classification

### ISSUE THREAD

**Issue by ahmad1368** [ahmad1368/GrowthPilotAI#25]
📝 Context
To improve the robustness of the "Auto-Scan" feature, we need a binary classifier that determines if the captured image is a Receipt/Invoice or Non-Document. Running OCR on a random photo is a waste of resources. We will use a lightweight, pre-trained TFLite model to "gatekeep" the OCR engine, ensuring it only processes valid financial documents.

👤 User Story
As a user, I want the app to quickly tell me if the photo I took isn't a receipt, so that I don't wait for the OCR to fail on a blurry or irrelevant image.

⚙️ Technical Requirements (Flutter Implementation)
1. TFLite Integration
Package: tflite_flutter: ^0.10.1 (or the latest FFI-based binding).

Model Assets: Add a quantized .tflite model (e.g., a modified MobileNetV2 or a custom Document-vs-Background model) to the assets/models/ directory.

Label Mapping: A simple labels.txt containing:

Plaintext
0 Receipt
1 Background
2. Image Pre-processing (The "Tensor" Prep)
Before the model can "see" the image, it must be resized and normalized:

Resize: Typically 224x224 or 299x299 pixels depending on the model.

Normalization: Convert pixel values from 0-255 to a float range of 0.0-1.0 or -1.0 to 1.0.

3. Classification Logic
Threshold: Only proceed to OCR (Issue #22) if the "Receipt" confidence score is > 0.70.

Feedback: If the score is low, trigger a Haptic feedback and a "Snackbar" saying: "Document not detected. Please align the receipt."

🛠 Logic & Code Guidance
The OCRService should be updated to include a "Validation" step before the heavy extraction starts.

Dart
import 'package:tflite_flutter/tflite_flutter.dart';

class DocumentClassifier {
  late Interpreter _interpreter;

  Future<void> loadModel() async {
    _interpreter = await Interpreter.fromAsset('models/receipt_classifier.tflite');
  }

  Future<bool> isReceipt(File imageFile) async {
    // 1. Convert File to Image and Resize (using 'image' package)
    // 2. Convert to Float32List (Tensor input)
    // 3. Run Inference
    var input = _preprocess(imageFile);
    var output = List.filled(2, 0).reshape([1, 2]); // Example for 2 classes
    
    _interpreter.run(input, output);
    
    // Check confidence for index 0 (Receipt)
    double confidence = output[0][0];
    return confidence > 0.75;
  }
}
✅ Acceptance Criteria (Definition of Done)
[ ] TFLite Model: The model is successfully bundled in the app assets and loaded by the Interpreter.

[ ] Fast Inference: The classification happens in < 100ms on mid-range devices.

[ ] Validation: The app correctly rejects non-financial images (e.g., a photo of a computer screen).

[ ] State Management: The UI shows a "Checking image quality..." state during the classification phase.

[ ] Resource Cleanup: The Interpreter is closed when the scanning session ends to free up GPU/CPU memory.

🔐 Security & Compliance
Offline Privacy: No images are sent to Google Cloud or TensorFlow Serving; all "judgment" happens on the local processor.

Model Size: Ensure the model is Quantized (likely < 5MB) to keep the app's total download size small for Canadian users on limited mobile data plans.

---
## 📱 Platform Requirements
Any code written to implement this issue must be delivered across all three supported platforms:
1. **Android** — implemented for the mobile Android build.
2. **iOS** — implemented for the mobile iOS build.
3. **Web** — implemented for the web build.

**Comment by ahmad1368** [comment]
Implemented and merged to `stage` in #775.
